ouroboros-consensus-cardano-test/src/Test/Consensus/Cardano/Examples.hs

```diff
@@ -71,29 +73,77 @@ eraExamples =
+      , exampleAnnTip           = viaInject Golden.exampleAnnTip
+      , exampleLedgerConfig     = exampleLedgerConfigCardano
+      , exampleLedgerState      = viaInject Golden.exampleLedgerState
+      , exampleChainDepState    = coerce (viaInject (coerce . Golden.exampleChainDepState) :: Labelled (WrapChainDepState (CardanoBlock Crypto)))
+      , exampleExtLedgerState   = viaInject Golden.exampleExtLedgerState
+      }
   where
-    eraInjections :: NP (Examples -.-> K (Examples (CardanoBlock Crypto)))
-                        (CardanoEras Crypto)
-    eraInjections =
-           fn (K . injExamples "Byron"   IZ)
-        :* fn (K . injExamples "Shelley" (IS IZ))
-        :* fn (K . injExamples "Allegra" (IS (IS IZ)))
-        :* fn (K . injExamples "Mary"    (IS (IS (IS IZ))))
-        :* fn (K . injExamples "Alonzo"  (IS (IS (IS (IS IZ)))))
-        :* Nil
-
-    injExamples ::
-         String
-      -> Index (CardanoEras Crypto) blk
-      -> Examples blk
-      -> Examples (CardanoBlock Crypto)
-    injExamples eraName idx =
-          Golden.prefixExamples eraName
-        . inject exampleStartBounds idx
+    viaInject :: forall f. Inject f => (forall blk. Examples blk -> Labelled (f blk)) -> Labelled (f (CardanoBlock Crypto))
+    viaInject getExamples =
+          coerce
+        $ mconcat
+        $ hcollapse
+        $ himap (\ix -> K . inj ix . getExamples) perEraExamplesPrefixed
+      where
+        inj :: forall blk. Index (CardanoEras Crypto) blk -> Labelled (f blk) -> Labelled (f (CardanoBlock Crypto))
+        inj idx = fmap (fmap (inject exampleStartBounds idx))
+
+    perEraExamplesPrefixed :: NP Examples (CardanoEras Crypto)
+    perEraExamplesPrefixed = hzipWith (\(K eraName) es -> Golden.prefixExamples eraName es) perEraNames perEraExamples
+      where
+        perEraNames = K "Byron"
+                   :* K "Shelley"
+                   :* K "Allegra"
+                   :* K "Mary"
+                   :* K "Alonzo"
+                   :* Nil
+
+    exampleLedgerConfigCardano :: Labelled (HardForkLedgerConfig (CardanoEras Crypto))
+    exampleLedgerConfigCardano = [
+        ( Nothing
+        , HardForkLedgerConfig
+          cardanoShape
+          (PerEraLedgerConfig (
+               WrapPartialLedgerConfig (ByronPartialLedgerConfig   lcByron   (TriggerHardForkAtEpoch shelleyTransitionEpoch))
+            :* WrapPartialLedgerConfig (ShelleyPartialLedgerConfig lcShelley (TriggerHardForkAtEpoch (History.boundEpoch allegraStartBound)))
+            :* WrapPartialLedgerConfig (ShelleyPartialLedgerConfig lcAllegra (TriggerHardForkAtEpoch (History.boundEpoch maryStartBound)))
+            :* WrapPartialLedgerConfig (ShelleyPartialLedgerConfig lcMary    (TriggerHardForkAtEpoch (History.boundEpoch alonzoStartBound)))
+            :* WrapPartialLedgerConfig (ShelleyPartialLedgerConfig lcAlonzo  TriggerHardForkNever)
+            :* Nil))
+        )
+      | lcByron   <- labelledLcByron
```

Comment:
This looks like a cardinality explosion. I don't know if there's a sensible alternative, though.

Reply:
Humm I'll add an assertion on the resulting cardinality and error out with a message to fix the test... since this is just used in tests and the cardinality is fixed, I think a simple error is appropriate.

Reply:
Sounds reasonable.

- Only idea that might improve it: let's move the maximum size check into the definition of `Test.Util.Serialisation.Golden.goldenTests` and have it as a test failure (aborting the rest of that invocation) instead of an `error`.

- Also, let's write the check such that it doesn't compute the whole list length (which could be billions if something went wrong) and instead requires `null (drop maxSize examples)`.